### PR TITLE
Add conversion for badger2 datastore

### PR DESCRIPTION
**Pull request by gammazero**
This change adds functionality to convert an existing datastore to/from a badger2 data store.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# ipfs can generate profiling dump files
```
Comment (aschmahmann) [inline .gitignore:1]: Adding a .gitignore file makes sense, thanks.

A couple things:
1) It should be in its own commit (or PR)
2) This looks like it was taken from go-ipfs which means we should both change the comment here and make it a bit more targeted to the types of outputs that tend to show up in this repo

File: go.mod [hunk 2]
```diff
@@ -3,16 +3,28 @@ module github.com/ipfs/ipfs-ds-convert
 go 1.12
```
Comment (aschmahmann) [inline go.mod:3]: Can you also add in a commit that bumps the minimum go version to 1.14 (to match what's in CI and go-ipfs)?

File: go.mod [hunk 3]
```diff
@@ -3,16 +3,28 @@ module github.com/ipfs/ipfs-ds-convert
...
 
 require (
-	github.com/ipfs/go-datastore v0.4.4
+	github.com/DataDog/zstd v1.4.5 // indirect
```
Comment (aschmahmann) [inline go.mod:6]: As with https://github.com/ipfs/go-ds-badger2/pull/31#discussion_r494323001 I think we should keep the go.mod file as clean as possible by just updating the dependencies we need.

File: go.mod [hunk 4]
```diff
@@ -3,16 +3,28 @@ module github.com/ipfs/ipfs-ds-convert
...
+	github.com/ipfs/go-ds-badger2 v0.1.0
+	github.com/ipfs/go-ds-flatfs v0.4.5
 	github.com/ipfs/go-ds-leveldb v0.4.2
 	github.com/ipfs/go-ds-measure v0.1.0
-	github.com/ipfs/go-fs-lock v0.0.4
-	github.com/ipfs/go-ipfs v0.5.0-rc2
-	github.com/ipfs/go-ipfs-config v0.5.2
+	github.com/ipfs/go-fs-lock v0.0.6
+	github.com/ipfs/go-ipfs v0.7.0
+	github.com/ipfs/go-ipfs-config v0.9.0
```
Comment (aschmahmann) [inline go.mod:20]: Let's have a commit that updates the go mod to the latest version of go-ipfs before we do the badger update

File: repo/badgerds.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
+package repo
+
```
Comment (aschmahmann) [inline repo/badgerds.go:2]: This is IMO a good refactor, but should be done separately from the badger2 update to make reviewing easier and allowing CI to run on changes more independently.

File: testfiles/badger2Spec [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (aschmahmann) [inline testfiles/badger2Spec:1]: Unless I'm mistaken it doesn't look like you've used this test file anywhere.

**Review by aschmahmann — CHANGES_REQUESTED** [review]
WDYT about splitting this up into 3 smaller PRs

1) A refactor/cleanup PR that adds the .gitignore (one commit) and moves around the datastore config functions
2) Updates go.mod for go-ipfs v0.7.0 (including the Go version)
3) Badger2 conversion

File: testfiles/badger2Spec (hunk 6 above)
Comment (aschmahmann) [reply testfiles/badger2Spec:1]: Note: If you were to actually run tests end-to-end tests using badger2 like this it would fail because there is not yet a badger2 plugin for go-ipfs. Once you have a draft PR up for that you can link it into this one and tests will be able to pass

Comment (aschmahmann) [reply testfiles/badger2Spec:1]: I don't see offhand any natural place to put a badger2 test. I think we should add a set of tests that just tries converting from all primitive datastores (flatfs, leveldb, memory, badger, badger2) to in memory leveldb and then back again.

**Comment by gammazero** [comment]
Closing to split into smaller/cleaner PRs
